File: projekt_2/gra/resources/Zwierzeta/Krolik.java
package gra.resources.Zwierzeta;

import gra.resources.backend.*;

public class Krolik extends Zwierze {
    private final static int KROLIK_SILA = 2;
    private final static int KROLIK_INIC = 7;
    private final static char KROLIK_CHAR = 'K';
    private final static int KROLIK_POLOG = 2;
    private final static int KROLIK_PELNOLETNOSC = 3;
    private final static String KROLIK_NAZWA = "Krolik";

    public Krolik(Swiat swiat, int x, int y) { super(swiat, x, y, KROLIK_SILA, KROLIK_INIC, KROLIK_PELNOLETNOSC); }

    public boolean Koliduj(Organizm somsiad, boolean czyAtakuje) {
        return false;
    }
    public void dodajPolog() { this.polog = KROLIK_POLOG; }
    public Organizm dajPotomka(Wsp n_wsp) {
        Organizm tmp = new Krolik(swiat, n_wsp.x, n_wsp.y);
        swiat.DodajOrganizm(tmp);
        return tmp;
    }
    public boolean czyTenSamGatunek(Organizm o) { if (o instanceof Krolik) { return true; } else return false; }
    public char getChar() { return KROLIK_CHAR; }
    public String getNazwa() { return KROLIK_NAZWA; }
}

File: projekt_1/Projekt_1/Projekt_1/Krolik.cpp
#pragma once
#include "Krolik.h"
#include <iostream>
using namespace std;

Krolik::Krolik(Swiat* swiat, int x, int y) : Zwierze(swiat, x, y, KROLIK_SILA, KROLIK_INIC, KROLIK_PELNOLETNOSC) {}

bool Krolik::Koliduj(Organizm* somsiad, bool czyAtakuje) {
	return false;
}

void Krolik::dodajPolog() { this->polog = KROLIK_POLOG; }
Organizm* Krolik::dajPotomka(Wsp n_wsp) {
	Organizm* tmp = new Krolik(swiat, n_wsp.x, n_wsp.y);
	swiat->DodajOrganizm(tmp);
	return tmp;
}
bool Krolik::czyTenSamGatunek(Organizm* o) { if (o->getChar() == KROLIK_CHAR) { return true; } else return false; }

char Krolik::getChar() { return KROLIK_CHAR; }
string Krolik::getNazwa() { return KROLIK_NAZWA; }
